### DISCUSSION TITLE
analogMeter with non-zero scale start

### DISCUSSION THREAD

**Discussion by jmuellers** [Bodmer/TFT_eSPI#2171] [Q&A - General]
Hello,
when using analogMeter with non-zero start value, it does not behave as expected.

```
tw.analogMeter(0, 0, 40.0, "°C", "0.0", "10.0", "20.0", "30.0", "40.0");
tw.updateNeedle(20.0, 0);
```
places the needle as expected.

```
tw.analogMeter(0, 0, 40.0, "°C", "10.0", "20.0", "30.0", "40.0", "50.0");
tw.updateNeedle(20.0, 0);
```
does not place the needle as expected.

How to treat this situation?
Is it not foreseen?

**Comment by Bodmer** [comment]
The meter needle always moves from 0.0 to the fullscale value (40.0 in this case). So the needle position is set be using a value from 0.0 to 40.0.

In the examples there is a floaring point mapValue(...) function that can be used to map 10.0 to 50.0 into the a new value in the range 0.0 to 40.0, this new value is used to update the meter needle.  In your use case it would be easier to simply subtract 10.0 and pass to the meter function.

**Reply by jmuellers** [reply]
Thank you for providing an answer. I expected the analogMeter to take care of the conversion, since all necessary parameters are (in principle) provided. I will use it as you suggested and subtract the offset.

**Reply by Bodmer** [reply]
The function analogMeter only specifies that full scale range (40.0 in this case), so cannot map the value to another range, the other parameters are just text labels so could be text ("Hot", "Warm", "Cold" etc.) with no numerical value associated.

I could add a new feature but am busy with other stuff at the moment!

**Reply by jmuellers** [reply]
Thank you again, it's a great and helpful library in any case :-)
